debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java

```diff
@@ -374,18 +385,28 @@ protected void handleServerHeartbeat(Event event) {
     /**
      * Handle the supplied event that signals that an out of the ordinary event that occurred on the master. It notifies the slave
      * that something happened on the master that might cause data to be in an inconsistent state.
-     * 
+     *
      * @param event the server stopped event to be processed; may not be null
      */
     protected void handleServerIncident(Event event) {
-        logger.trace("Server incident: {}", event);
+        if (event.getData() instanceof EventDataDeserializationExceptionData) {
+            EventDataDeserializationExceptionData data = event.getData();
+
+            logger.error(
+                    String.format("Exception during event data deserialization (source info: %s, partion: %s)", source, source.partition()),
+                    data.getCause()
+            );
+        }
+        else {
+            logger.warn("Server incident: {}", event);
```

Comment:
Yes, I'll change it to ERROR. It used to be TRACE before which didn't seem right, so I changed it to WARN. But ERROR is better indeed. As to when it can happen, the docs say that: "something happened on the master that might cause data to be in an inconsistent state". But we'll handle them as before (just logging), only our special data will cause the special logging with more details.